ThunderTable/InputPickerRow.swift

```diff
@@ -0,0 +1,182 @@
+
+/// A protocol which can be used to represent a component in a `UIPickerView`
+public protocol PickerComponentDisplayable {
+	
+	/// The items to be displayed in the component
+	var items: [PickerRowDisplayable] { get }
+}
+
+/// A helper class for ease of use of `PickerComponentDisplayable`
+open class PickerComponent: PickerComponentDisplayable {
+	
+	/// The items to be displayed in the component
+	open var items: [PickerRowDisplayable]
+	
+	public init(items: [PickerRowDisplayable]) {
+		self.items = items
+	}
+}
+
+/// A protocol which can be used to provide rows to a `UIPickerView` component
+public protocol PickerRowDisplayable {
+	
+	/// The value that this row represents, doesn't have to be the same as title
+	var value: AnyHashable { get }
+	
+	/// The title to display on this row
+	var rowTitle: String { get }
+	
+	/// The attributed title to display on this row (Optional)
+	var attributedTitle: NSAttributedString? { get }
+}
+
+public extension PickerRowDisplayable {
+	
+	public var attributedTitle: NSAttributedString? { get { return nil } }
+}
+
+/// A helper class for ease of use of `PickerRowDisplayable`
+open class PickerRow: PickerRowDisplayable {
+	
+	/// The value that this row represents, doesn't have to be the same as title
+	open var value: AnyHashable
+	
+	/// The title to display on this row
+	open var rowTitle: String
+	
+	/// Creates a new instance with a title and value
+	///
+	/// - Parameters:
+	///   - title: The title to display in the `UIPickerView`
+	///   - value: The value that the row represents
+	public init(title: String, value: AnyHashable) {
+		self.rowTitle = title
+		self.value = value
+	}
+}
+
+/// A row which displays a picker in the keyboard for the user
+/// to select from a set of defined options
+open class InputPickerRow: InputTableRow {
+	
+	/// The components to display in the picker
+	open var components: [PickerComponentDisplayable]
+	
+	/// A closure to format the value of the row
+	open var formatter: ((_ value: [AnyHashable]) -> String)
+	
+	/// Creates a new row from provided properties
+	///
+	/// - Parameters:
+	///   - title: The title of the row
+	///   - mode: The mode of the date picker
+	///   - id: The unique id for this row
+	///   - required: Whether the value is required
+	public init(title: String?, components: [PickerComponentDisplayable], formatter: @escaping ((_ value: [AnyHashable]) -> String), id: String, required: Bool) {
+		
+		self.formatter = formatter
+		self.components = components
+		super.init(id: id, required: required)
+		self.title = title
+	}
+	
+	open override var cellClass: AnyClass? {
+		return InputPickerViewCell.self
+	}
+	
+	private var cellTextField: UITextField?
+		
+	open override func configure(cell: UITableViewCell, at indexPath: IndexPath, in tableViewController: TableViewController) {
+		
+		guard let pickerCell = cell as? InputPickerViewCell else { return }
+		
+		super.configure(cell: cell, at: indexPath, in: tableViewController)
```

Comment:
Would you not want to call super before the guard? Seems strange to just return without calling super if the cell isn't our subclass type.